### PR TITLE
Support renaming indexes

### PR DESCRIPTION
**Pull request by drigberg**
CHANGES
- `RENAME INDEX` support
- Tests

### CODE REVIEW

File: tests/03-indexes.sql [hunk 1]
```diff
@@ -6,3 +6,5 @@ CREATE TABLE aaa (
 
 CREATE INDEX foo_idx ON aaa (user_id);
 -- CREATE INDEX bar_idx ON aaa (user_id, type);
+
+ALTER TABLE aaa RENAME INDEX foo_idx TO foobar_idx;
```
Comment (nvie) [inline tests/03-indexes.sql:10]: The test suite will run the query against an _actual_ test DB with MySQL and will compare the final output to the simulator, to ensure 100% accuracy.

Can you also please add a test case for renaming an "implicitly created" index? MySQL is particularly gnarly around those edge cases! For example, if you create a foreign key first, then rename the implicitly-created index it creates along with that. Sometimes the names auto-change along with the FKs, but in other cases they don't.

Lastly, would you mind putting this test in a new test case instead of adding it to this test file?

**Review by nvie — COMMENTED** [review]
Nooice! And thank you for your first contribution here! 🙌 ❤️

File: tests/03-indexes.sql (hunk 1 above)
Comment (drigberg) [reply tests/03-indexes.sql]: Done!

I don't see any implicit index to rename for foreign keys. I added a foreign key to this test, but it only adds a foreign key, which can't be renamed this way:

```
CREATE TABLE `aaa` (
  `id` int(11) DEFAULT NULL,
  `bbb_id` int(11) DEFAULT NULL,
  `type` varchar(16) NOT NULL,
  UNIQUE KEY `bbb_id_type_uniq2` (`bbb_id`,`type`),
  KEY `foobar_idx` (`type`),
  CONSTRAINT `aaa_ibfk_1` FOREIGN KEY (`bbb_id`) REFERENCES `bbb` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

CREATE TABLE `bbb` (
  `id` int(11) NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

```

Comment (drigberg) [reply tests/03-indexes.sql]: Ex: 
`Error: Index ... does not exist on table "aaa". These do: foobar_idx, bbb_id_type_uniq2`

Comment (nvie) [reply tests/03-indexes.sql]: Yeah, exactly, this is part of the issue. If you run the following in MySQL directly (**not** in the simulator):

```sql
CREATE TABLE bbb (
  id INT PRIMARY KEY
);

CREATE TABLE aaa (
  id INT,
  bbb_id INT,
  type VARCHAR(16) NOT NULL,
  FOREIGN KEY (bbb_id) REFERENCES bbb (id)
);

ALTER TABLE aaa RENAME INDEX bbb_id TO bbb_id2;
```

Then, you'll notice that the `bbb_id` implicitly-created index does exist, and will appropriately get renamed. You can inspect the results that MySQL produces:

```
SHOW CREATE TABLE aaa;
/*
CREATE TABLE `aaa` (
  `id` int(11) DEFAULT NULL,
  `bbb_id` int(11) DEFAULT NULL,
  `type` varchar(16) NOT NULL,
  KEY `bbb_id2` (`bbb_id`),  -- <=== NOTE how this got renamed!
  CONSTRAINT `aaa_ibfk_1` FOREIGN KEY (`bbb_id`) REFERENCES `bbb` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8
*/
```

So the problem here is that the MySQL simulator chokes on this valid SQL, which is a bug.

I realize this testing setup is rather crude — sorry!

Comment (drigberg) [reply tests/03-indexes.sql]: Figured it out! `UNIQUE KEY `bbb_id_type_uniq2` (`bbb_id`,`type`),` was overwriting the implicit index from the foreign key, bc mysql magic. If you create the tables without that unique constraint, the `bbb_id` key magically reappears!

**Comment by nvie** [comment]
W00t! Glad you figured it out and found out about MySQL magic! Nice work, dude :) Thanks for the awesome addition to mysql-simulator 🎉

**Review by nvie — APPROVED** [review]
